service/src/main/java/com/codeforcommunity/requester/MapRequester.java

```diff
@@ -41,85 +46,132 @@ public MapRequester(Vertx vertx) {
-
-    Future<Void> updateFuture =
+    Future<JsonArray> getBlockFIDsFuture =
         tokenFuture.compose(
             tokenString -> {
               MultiMap formData =
                   MultiMap.caseInsensitiveMultiMap()
                       .add("f", "json")
                       .add("token", tokenString)
-                      .add("updates", updateJson.encode());
+                      .add("where", "ID IN (" + String.join(",", blockIds) + ")")
+                      .add("outFields", "FID");
 
               return Future.future(
                   promise -> {
                     client
-                        .postAbs(featureLayerRoute)
+                        .postAbs(featureLayerQueryRoute)
                         .putHeader("content-type", "multipart/form-data")
                         .sendForm(
                             formData,
-                            (ar) -> {
-                              if (ar.succeeded()) {
-                                HttpResponse<Buffer> httpResponse = ar.result();
+                            (asyncResult -> {
+                              if (asyncResult.succeeded()) {
+                                HttpResponse<Buffer> httpResponse = asyncResult.result();
 
                                 if (httpResponse.statusCode() == 200) {
                                   JsonObject responseBody = httpResponse.bodyAsJsonObject();
-                                  if (responseBody.containsKey("error")
-                                      && responseBody.getJsonObject("error").getInteger("code")
-                                          == 498) {
-                                    // The API token is invalid, reset it and make this call again.
-                                    logger.info(
-                                        "Remaking token request while updating blocks to "
-                                            + updateTo.name());
-                                    this.tokenFuture = updateToken();
-                                    updateLayers(streetFids, updateTo, this.tokenFuture)
-                                        .onSuccess((V) -> promise.complete())
-                                        .onFailure(promise::fail);
-                                  } else if (responseBody.containsKey("updateResults")) {
-                                    logger.info(
-                                        "Successfully updated blocks to " + updateTo.name());
-                                    // Check successes
-                                    promise.complete();
-                                  } else {
-                                    // throw error
-                                    logger.error(
-                                        "Update street request responded with unrecognized response body: "
-                                            + responseBody);
+                                  if (responseBody.containsKey("features")) {
```

Comment:
Not sure if I'm just reading this wrong, but if we get a `200` that doesn't contain `features`, we do nothing? Do we have an error log for this?

Reply:
I believe you are correct, there's more to be done with logging/error handling. For example, all of my requests succeeded the whole time, but weren't actually. 

Pretty sure the real issue is that the API returns a 200 before this even completes... @liammoyn is that true? 

I'm going to write a method to validate that the features are "successful" updates and error if not. 

We need this code to be 100% clear and solid - every map request calls this now.

Reply:
This is checking that the JSON body is what we expect. If this is not present then we can't continue the function because we don't know where the data is. I believe the correct behavior is to just log the response body that we got along with an error like 'missing features array in response body'